app/components/course/CourseDropDownButton.tsx

```diff
@@ -1,41 +1,82 @@
 import React, { useEffect, useState } from 'react';
+import { CourseCardProps } from "@/app/model/CourseInterface";
+import { useAppContext } from "../../contexts/appContext/AppProvider";
+import { SingleCourse } from "@/app/model/CourseInterface";
 
-const CourseCardDropDown = () => {
-  const [isOpen, setIsOpen] = useState(false);
-  const [dropDownText, setDropDownText] = useState("No Selection");
+const colorMap = {
+  "No Selection": "bg-primary-700 text-white",
+  "In Progress": "bg-orange-500 text-white",
+  "Completed": "bg-green-500 text-white",
+  "Planned": "bg-gray-500 text-white", 
+};
 
-  const handleMouseEnter = () => {
-    setIsOpen(true);
-  };
+const CourseCardDropDown = ({
+  title,
+  courseCode,
+  status = "No Selection"
+}: CourseCardProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { coursesSelected, setCoursesSelected } = useAppContext();
+  status = coursesSelected.find(course => course.courseID === courseCode)?.status || "No Selection";
+  const [dropDownText, setDropDownText] = useState<string>(status);
+  
+  
 
-  const handleMouseLeave = () => {
-    setIsOpen(false);
-  };
+  const updateCourseInContext = (newState: string) => {
+    console.log(coursesSelected);
+    let to_add : SingleCourse = { courseID: courseCode, title: title, status: newState };
+    const updatedCourses = coursesSelected.map(course =>
+      course.courseID === courseCode
+        ? { ...course, status: newState }
+        : course);
+    let selectedIDS = updatedCourses.map(course => course.courseID);
 
-  const handleOption1 = () => {
-    setIsOpen(false);
-    setDropDownText("Planned");
-  };
-  const handleOption2 = () => {
-    setIsOpen(false);
-    setDropDownText("In Progress");
+    if (!selectedIDS.includes(to_add.courseID)) {
+      updatedCourses.push(to_add);
+    }
+    console.log(updatedCourses);
+    setCoursesSelected(updatedCourses);
   };
-  const handleOption3 = () => {
+  
+  const handleOption = (newStatus: string) => {
+    setDropDownText(newStatus);
+    updateCourseInContext(newStatus);
     setIsOpen(false);
-    setDropDownText("Interested");
   };
+  /*
+  useEffect(() => {
+    if (dropDownText !== "No Selection") {
+      setIsOpen(true);
+    }
+  }, [dropDownText]);
+  */
+  
 
   return (
-    <div className="relative" onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
-      <div className="text-sm font-semibold text-gray-25 bg-primary-700 px-2 py-2.5 border border-solid border-gray-300 rounded-lg cursor-pointer text-center w-28">
+    <div
+      className="relative"
+      onMouseEnter={() => setIsOpen(true)}
+      onMouseLeave={() => setIsOpen(false)}
+    >
+      <div
+        id="fixed-size-div"
+        className={`text-sm font-semibold px-2 py-2.5 border border-solid border-gray-300 rounded-lg cursor-pointer text-center ${colorMap[dropDownText]}`}
```

Comment:
use clsx package for a dynamic class. that way Tailwind doesn't have to create all classes to handle.

Reply:
Fixed.